packages/core/src/index.ts

```diff
@@ -392,28 +401,28 @@ class Hopr extends EventEmitter {
    * Called whenever a peer is announced
    * @param peer newly announced peer
    */
-  private onPeerAnnouncement(peer: { id: PeerId; multiaddrs: Multiaddr[] }): void {
+  private async onPeerAnnouncement(peer: { id: PeerId; multiaddrs: Multiaddr[] }): Promise<void> {
```

Comment:
Since this is bound to an event a line 299, what happens if this method throws exception during async execution? Seems that it will be a source of unhandled rejection coming from event manager.